packages/atclient/src/atsign.c

```diff
@@ -1,13 +1,37 @@
-#ifndef ATCLIENT_ATSIGN_H
-#define ATCLIENT_ATSIGN_H
-
 #include "atclient/atsign.h"
 #include "atlogger/atlogger.h"
 #include <stdio.h>
 #include <string.h>
 
 #define TAG "atsign"
 
+int atclient_atsign_init(atclient_atsign *atsign, const char *atsign_str) {
+  int ret = 0;
+
+  // atsign_str is longer than expected or null/empty
+  if ((strlen(atsign_str) > MAX_ATSIGN_STR_BUFFER) || (atsign_str == NULL) || (strlen(atsign_str) == 0)) {
+    ret = 1;
+    atclient_atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "atclient_atsign_init: %d\n", ret);
+    return ret;
+  }
+
+  memset(atsign, 0, sizeof(atclient_atsign));
+  atsign->atsign = malloc(strlen(atsign_str) + 1);
+
+  const unsigned long maxatlen = MAX_ATSIGN_STR_BUFFER;
+  unsigned long atolen = 0;
+  ret = atclient_atsign_with_at_symbol(atsign->atsign, maxatlen, &(atolen), atsign_str, strlen(atsign_str));
+  if (ret != 0) {
+    atclient_atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "atclient_atsign_with_at_symbol failed\n");
+    return ret;
+  }
+
+  atsign->without_prefix_str = atsign->atsign + 1;
+  return ret;
+}
+
+void atclient_atsign_free(atclient_atsign *atsign) { free(atsign->atsign); }
+
 int atclient_atsign_without_at_symbol(char *atsign, const unsigned long atsignlen, unsigned long *atsignolen,
```

Comment:
do we still need this?

Reply:
Not right now, but could be useful. We can always remove it for now and put it back if we need it

Reply:
Ohh actually it is used in `atclient_pkam_authenticate`:
https://github.com/atsign-foundation/at_c/blob/18a4357e07887c4cc5a38cb78f665592290f0052/packages/atclient/src/atclient.c#L109